grep-00XX-relicense-volk.md

```diff
@@ -0,0 +1,86 @@
+Previously, several attemps were made to relicense VOLK without success.
+With a formal process and a clear vision laid out, we hope to be able to get everyone on board.
+
+Since VOLK is a library it should choose a fitting license.
+If we see greater adoption of VOLK in more project, we hope to also see more contributions.
+May it be bug fixes, new kernels or even contributions to core features.
+
+## Description
+
+In order to facilitate this relicensing effort, we need to got through several steps.
+
+### Notification
+We consider it to be very important to let every contributor know that the wish to relicense by some contributors exists.
+A first step would be to notify every contributor of this GREP.
+
+### Another license discussion
+Currently, the proposed license is LGPLv3+.
+Though, other licenses, possibly more permissive are available too.
+This includes [3-Clause BSD](https://opensource.org/licenses/BSD-3-Clause), [MIT](https://opensource.org/licenses/MIT) and [Apache-2.0](https://www.apache.org/licenses/LICENSE-2.0).
+Prelimenary discussion on the subject matter revealed that Apache-2.0 is out of favor by some.
+We assume that LGPLv3+ strikes a balance between permissiveness and obligation to provide modified source code.
+
+### All contributors need to sign off for the new license
+
+Before we can adopt a new license, we need approval by every contributor to basically resubmit their contributions under the new license.
+[There is more to this discussion and how this evolved but others are more qualified to add to that discussion]
+
+### The process
+We need to figure out how we collect all the contributor approvals.
+This might be a file within the VOLK repo where everyone adds a signed-off commit to approve the new license.
+Currently, we have around 65 contributors to VOLK.
+
+### The meantime
+It might take time to gain approval by every contributor.
+We want to be able to continue VOLK development while we are in this license transition period.
+We need to work out a clear path how to deal with contributions in the mean time.
+This will probably prior include relicensing approval by a majority of contributors.
+Then, we will probably state that new contributions will only be accepted under the terms of the new license.
+The details need to be worked out.
+
+### The aftermath
+Assume most contributors approved to relicense VOLK but some did not.
+How do we deal with that situation?
+A contributor who does not approve the new license, cannot be forced or pushed to do so.
+Either, we decide to give up on this relicensing effort or, we are required to remove all contributions by that contributor.
+Though, we hope to avoid this situation.
+Yet, we need to define a final date by which we will end the transition period.
+Otherwise, we risk to stay in relicensing limbo.
```

Comment:
Maybe say that you'll also want to do a major release at that point, simply to make that distinction easy for consumers of the library. E.g., if you've got non-GPL code that wants to use VOLK, `find_package(VOLK, >=14.0)`, simple as that.

Maybe defining a "last version number that also includes optionally enabled GPL kernels" is better than a date?

Reply:
First, I thought, we'd bump this to a new major version number. This might be the case if we need to drop kernels entirely. Otherwise, I'd probably like to stick with the VOLK 2.x line.
How does a license change work with packaging? @maitbot Do we need to consider anything here?

Reply:
I definitely wouldn't do this within the 2.x line.

Reply:
> Maybe defining a "last version number that also includes optionally enabled GPL kernels" is better than a date?

A date still works if we require all contributions to any VOLK branch to be LGPL, or relicensable under LGPL.

Reply:
I really don't have an idea how long it'll take for most contributors to react to such an email and take action. Based on our experience after a short while, we can make a good statement.
Alternatively, we can set a date e.g. 2 months into the future. Then we follow the best academia tradition and extend this deadline.